src/plugins/techStackGeneric.nim

```diff
@@ -0,0 +1,408 @@
+        if contains(categories[category], subcategory):
+            categories[category][subcategory].add(key)
+        else:
+            categories[category][subcategory] = @[key]
+    else:
+        categories[category] = newTable[string, seq[string]]()
+        categories[category][subcategory] = @[key]
+
+    if val.hostScope != nil:
+        if not contains(inHostScope, category):
+            inHostScope[category] = newTable[string, bool]()
+        inHostScope[category][subcategory] = false
+    else:
+        if val.fileScope == nil:
+            error("One of file_scope, host_scope must be defined for rule " & key & ". Skipping")
+            continue
+        if not contains(inFileScope, category):
+            inFileScope[category] = newTable[string, bool]()
+        inFileScope[category][subcategory] = false
+
+        tsRules[key] = val
+        regexes[key] = re(val.fileScope.getRegex())
+        headLimits[key] = val.fileScope.getHead()
+        let filetypes = val.fileScope.getFiletypes()
+        if filetypes.isSome():
+            let ftypes = filetypes.get()
+            ruleFiletypes[key] = ftypes
+            for ft in ftypes:
+                if contains(ftRules, ft):
+                    ftRules[ft].incl(key)
+                else:
+                    ftRules[ft] = toHashSet([key])
+        else:
+            # we only have exclude rules therefore we match by default
+            # XXX move to a template for looking things up and adding if
+            # they don't exist
+            if contains(ftRules, FT_ANY):
+                ftRules[FT_ANY].incl(key)
+            else:
+                ftRules[FT_ANY] = toHashSet([key])
+            let excludeFiletypes = val.fileScope.getExcludedFiletypes()
+            if excludeFiletypes.isSome():
+                let exclFtps = excludeFiletypes.get()
+                ruleExcludeFiletypes[key] = exclFtps
+                for ft in exclFtps:
+                    if contains(excludeFtRules, ft):
+                        excludeFtRules[ft].incl(key)
+                    else:
+                        excludeFtRules[ft] = toHashSet([key])
+
+        # get paths and excluded paths that need to always be considered
+        let filepaths = val.fileScope.getFilepaths()
+        if filepaths.isSome():
+            let fpaths = filepaths.get()
+            for path in fpaths:
+                if contains(pthRules, path):
+                    pthRules[path].incl(key)
+                else:
+                    pthRules[path] = toHashSet([key])
+
+        let excludeFilepaths = val.fileScope.getExcludedFilepaths()
+        if excludeFilepaths.isSome():
+            let excfpaths = excludeFilepaths.get()
+            for path in excfpaths:
+                if contains(excludePthRules, path):
+                    excludePthRules[path].incl(key)
+                else:
+                    excludePthRules[path] = toHashSet([key])
+
+  newPlugin("techStackGeneric",
```

Comment:
why indented? this should be top level in the module?

similar comment as above. can we use consistent whitespace across chalk

Reply:
actually this does need to be inside the load function